src/components/Tool/Appeal/Modals/AddExcludedContactModal/AddExcludedContactModal.tsx

```diff
@@ -0,0 +1,137 @@
+import React, { useState } from 'react';
+import {
+  Box,
+  CircularProgress,
+  DialogActions,
+  DialogContent,
+  DialogContentText,
+} from '@mui/material';
+import { styled } from '@mui/material/styles';
+import { useSnackbar } from 'notistack';
+import { useTranslation } from 'react-i18next';
+import {
+  CancelButton,
+  SubmitButton,
+} from 'src/components/common/Modal/ActionButtons/ActionButtons';
+import Modal from 'src/components/common/Modal/Modal';
+import {
+  AppealsContext,
+  AppealsType,
+} from '../../AppealsContext/AppealsContext';
+import { useAppealQuery } from '../AddContactToAppealModal/appealInfo.generated';
+import { useAssignContactsToAppealMutation } from './AddExcludedContactModal.generated';
+
+const LoadingIndicator = styled(CircularProgress)(({ theme }) => ({
+  margin: theme.spacing(0, 1, 0, 0),
+}));
+
+export interface AddExcludedContactModalProps {
+  contactIds: string[];
+  handleClose: () => void;
+}
+
+export const AddExcludedContactModal: React.FC<
+  AddExcludedContactModalProps
+> = ({ contactIds, handleClose }) => {
+  const { t } = useTranslation();
+  const { enqueueSnackbar } = useSnackbar();
+  const [assignContactsToAppeal] = useAssignContactsToAppealMutation();
+  const { accountListId, appealId, contactsQueryResult } = React.useContext(
+    AppealsContext,
+  ) as AppealsType;
+  const [mutating, setMutating] = useState(false);
+
+  const { data, loading } = useAppealQuery({
+    variables: {
+      accountListId: accountListId ?? '',
+      appealId: appealId ?? '',
+    },
+  });
+
+  const existingContactIds = data?.appeal?.contactIds ?? [];
+  const addingManyContacts = contactIds.length > 1;
+
+  const handleConfirm = async () => {
+    setMutating(true);
+
+    if (!contactIds.length) {
+      enqueueSnackbar(t('Failed to add contact(s) to appeal'), {
+        variant: 'error',
+      });
+      return;
+    }
+    await assignContactsToAppeal({
+      variables: {
+        input: {
+          accountListId: accountListId ?? '',
+          attributes: {
+            id: appealId,
+            contactIds: [...existingContactIds, ...contactIds],
+            forceListDeletion: true,
+          },
+        },
+      },
+      update: () => {
+        contactsQueryResult.refetch();
+      },
+      onCompleted: () => {
+        {
+          addingManyContacts
```

Comment:
Since the message is the only thing that changes, can you change this to `enqueueSnackbar(addingManyContacts ? message1 : message2, options)`? And `onError`.